### PR TITLE
ENH Add alt attribute for thumbnails

### LINKED ISSUE #664: ENH More informative 'alt' attribute for thumbnails in index
Following from #663, make the alternative text ('alt' attribute of image in rst) for thumbnail images in the index.rst of generated galleries to be the title of the example file.

cc @ogrisel

### PR DESCRIPTION
**Pull request by lucyleeow**
closes #664

### CODE REVIEW

**Comment by lucyleeow** [comment]
@larsoner I don't understand the cause of the test failures. I seems that `test_junit` and `test_rebuild` are failing during setup but I don't understand why. tinybuild builds fine when I check.
The error is
```
ValueError: not enough values to unpack (expected 3, got 2)
```
but there all 3 output variables (intro, title, cost) are present.

**Comment by larsoner** [comment]
https://github.com/sphinx-gallery/sphinx-gallery/pull/668/files#diff-0cc5ccb98f5854a18c184ad4644a3509R759

beware the multiple `return` code path...

**Comment by lucyleeow** [comment]
@larsoner the title cannot contain `\n` so we won't have the same problem as in #669. It can contain non-word characters but will they be a problem?
Also do you think the test is adequate?
And thanks for the tip!

File: sphinx_gallery/gen_rst.py [hunk 1]
```diff
@@ -336,12 +336,13 @@ def generate_dir_rst(src_dir, target_dir, gallery_conf, seen_backrefs):
         'generating gallery for %s... ' % build_target_dir,
         length=len(sorted_listdir))
     for fname in iterator:
-        intro, cost = generate_file_rst(
+        print('for {} generate file run'.format(fname))
```
Comment (larsoner) [inline sphinx_gallery/gen_rst.py:339]: debugging cruft?

**Comment by larsoner** [comment]
>  It can contain non-word characters but will they be a problem?

Maybe but we can wait until someone has a problem :)

File: sphinx_gallery/gen_rst.py (hunk 1 above)
Comment (lucyleeow) [reply sphinx_gallery/gen_rst.py:339]: :joy: oh no, my print statement debugging exposed!

**Comment by lucyleeow** [comment]
@larsoner added a check in html and rst for index and backref thumbnail alt. Also added a html test to image alts.

**Comment by larsoner** [comment]
Thanks @lucyleeow

**Comment by lucyleeow** [comment]
cc @ogrisel see https://1473-25860190-gh.circle-artifacts.com/0/rtd_html/auto_examples/index.html for thumbnail alt example :)

**Pull request merged**